src/main/java/servlets/Upload.java

```diff
@@ -0,0 +1,43 @@
+package servlets;
+
+import IO.WriterSimple;
+import encoder.Encoder;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import shared.sharedObjects;
+
+/** Fetch the given data points for a specific run based on its runId and the dataType. */
+@MultipartConfig
+@WebServlet("/upload")
+public class Upload extends HttpServlet {
+  // Return numEntries runId and name pairs for getting urls to load viewer data.
+  // Optional numEntries parameter limits the number of returned values.
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException, IllegalStateException {
+    response.setContentType("text/html;");
+
+    String dataType = "pose";
```

Comment:
You can get the data type from request.

Reply:
Yeah totally agree, for mvp I think we are assuming just pose data, but I'll change this when we handle other forms.